pkg/monitoring/poller.go

```diff
@@ -0,0 +1,70 @@
+package monitoring
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/gagliardetto/solana-go"
+)
+
+type Poller interface {
+	Start(context.Context)
+	Updates() <-chan interface{}
+}
+
+func NewPoller(
+	account solana.PublicKey,
+	reader AccountReader,
+	fetchInterval time.Duration,
+	bufferCapacity uint32,
+) Poller {
+	return &solanaPollerImpl{
+		account,
+		reader,
+		fetchInterval,
+		bufferCapacity,
+		make(chan interface{}, bufferCapacity),
+	}
+}
+
+type solanaPollerImpl struct {
+	account        solana.PublicKey
```

Comment:
This is the only part that makes this `Poller` implementation Solana specific, but we could easily generalize it. Consider just using string for account identifier:

```suggestion
type poller struct {
	account        string
```